pkg/api/types.go

```diff
@@ -330,6 +330,27 @@ type TestStepConfiguration struct {
 	// create a periodic job instead of a presubmit
 	Cron *string `json:"cron,omitempty"`
 
+	// PullRequestPolicy defines when this test is run as
+	// part of pull requests. It is a validation error to
+	// set this field when Cron is also set. Policies:
+	//
+	// * Always: this test is run on every pull request
+	// * OnRequest: this test will not be run until a user
+	//     explicitly requests it via /test NAME, at which
+	//     point it will be required for merge.
+	// * RunIfChanged: this test will be run if the regex
+	//     in pull_request_run_if_changed matches a change
+	//     in the PR.
+	//
+	// The default value is Always
+	PullRequestPolicy string `json:"pull_request_policy"`
```

Comment:
Is there a reason why these should not be `omitempty`? We would generate a ton of useless empty strings when determinizing and generating configs for release branches.